src/mediapipe_calculators/python_executor_calculator.cc

```diff
@@ -0,0 +1,160 @@
+//*****************************************************************************
+// Copyright 2023 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <unordered_map>
+#include "../python/ovms_py_tensor.hpp"
+#include "../mediapipe_internal/pythonnoderesource.hpp"
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#include "mediapipe/framework/calculator_framework.h"
+#pragma GCC diagnostic pop
+
+#include <Python.h>
+#include <pybind11/embed.h> // everything needed for embedding
+#include <pybind11/stl.h>
+
+#include "../python/python_backend.hpp"
+
+namespace py = pybind11;
+using namespace py::literals;
+using namespace ovms;
+
+namespace mediapipe {
+
+typedef std::unordered_map<std::string, std::shared_ptr<PythonNodeResource>> PythonNodesResources;
+
+const std::string INPUT_SIDE_PACKET_TAG = "PYTHON_NODE_RESOURCES";
+
+typedef std::unique_ptr<OvmsPyTensor> OvmsPyTensorPtr;
+
+class PythonExecutorCalculator : public CalculatorBase {
+
+/* 
+TODO: Streaming support:
+    - timestamping
+    - loopback input
+*/ 
+
+std::shared_ptr<PythonNodeResource> nodeResources;
+std::unique_ptr<PyObjectWrapper<py::iterator>> pyIteratorPtr;
+
+void pushOutputs(CalculatorContext* cc, py::list pyOutputs) {
+    py::gil_scoped_acquire acquire;
+    for (const std::string& tag : cc->Outputs().GetTags()) {
+        for (py::handle pyOutputHandle : pyOutputs) {
+            py::object pyOutput = pyOutputHandle.cast<py::object>();
+            nodeResources->pythonBackend->validateOvmsPyTensor(pyOutput);
+            if (pyOutput.attr("name").cast<std::string>() == tag) {
```

Comment:
Correct me if I'm wrong, but tags are reserved for input stream kind, meaning, all python outputs will have the same prefix, example:
```
input_stream: PYOBJECT1:my_input_a
input_stream: PYOBJECT2:my_output_b
output_stream: PYOBJECT3:my_output_a
output_stream: PYOBJECT4:my_output_b
```
Does this mean that we can only use input names starting with `PYOBJECT`?

Reply:
For Python we would have only one tag for all inputs and outputs being "OVMS_PY_TENSOR" and different stream names.
```
input_stream: OVMS_PY_TENSOR:my_input_a
input_stream: OVMS_PY_TENSOR:my_input_b
output_stream: OVMS_PY_TENSOR:my_output_a
output_stream: OVMS_PY_TENSOR:my_output_b
```

Reply:
But in the code you try to match python output names vs MediaPipe output stream **tags**, not names. All tags are the same. How does it work?

Reply:
You're right, tags will need to be different. The code works regardless though.